sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -376,6 +378,62 @@ def _affinity(self, other: "Provider"):
       return super()._affinity(other)
 
 
+@ExternalProvider.register_provider_type('yaml')
+class YamlProvider(Provider):
+  def __init__(self, transforms: Mapping[str, Mapping[str, Any]]):
+    if not isinstance(transforms, dict):
+      raise ValueError('Transform mapping must be a dict.')
+    self._transforms = transforms
+
+  def available(self):
+    return True
+
+  def cache_artifacts(self):
+    pass
+
+  def provided_transforms(self):
+    return self._transforms.keys()
+
+  def config_schema(self, type):
+    return json_utils.json_schema_to_beam_schema(self.json_config_schema(type))
+
+  def json_config_schema(self, type):
+    return dict(
+        type='object',
+        additionalProperties=False,
+        **self._transforms[type]['config_schema'])
+
+  def description(self, type):
+    return self._transforms[type].get('description')
+
+  def requires_inputs(self, type, args):
+    return self._transforms[type].get(
+        'requires_inputs', super().requires_inputs(type, args))
+
+  def create_transform(
+      self,
+      type: str,
+      args: Mapping[str, Any],
+      yaml_create_transform: Callable[
+          [Mapping[str, Any], Iterable[beam.PCollection]], beam.PTransform]
+  ) -> beam.PTransform:
+    from apache_beam.yaml.yaml_transform import SafeLineLoader, YamlTransform
+    spec = self._transforms[type]
+    try:
+      import jsonschema
+      jsonschema.validate(args, self.json_config_schema(type))
+    except ImportError:
+      pass
+    body = spec['body']
+    if not isinstance(body, str):
+      body = yaml.safe_dump(SafeLineLoader.strip_metadata(body))
+    templated = (  # keep formatting
+        jinja2.Environment(undefined=jinja2.StrictUndefined)
```

Comment:
This should probably use the same logic as main (i.e. allows GCS file includes)
```suggestion
        jinja2.Environment(undefined=jinja2.StrictUndefined, loader=_BeamFileIOLoader())
```
Maybe it would make sense to make a standalone function for loading jinja templates to help keep functionality synced

Reply:
Yeah, I'll do that.